### PR TITLE
docs: Add proper README

### LINKED ISSUE #49: Create documentation

### PR DESCRIPTION
**Pull request by pawfa**
Closes #49 
Created Readme with description and examples of all currently implemented features. Included some tricky cases which I found during pact-gen-ts configuration by developers unfamiliar to library.
All comments and ideas are welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,225 +169,72 @@ function fetchComments() {
...
+function fetchImage(imageId: number) {
     // ...
-    return data;
 }
 ```
 
-Pact-gen-ts recognizes response body type of interaction from returned object.
-
-## [x] Set manually what is response body
+`@pact-response-body` - sets expected body for current response, **IMPORTANT** - JSDoc has to be applied to separate variable - **not** directly to axios response
```
Comment (Fiszcz) [inline README.md:220]: I would write:
````
`@pact-response-body` - sets expected body for current response

```ts
/**
 * @pact
 */
async function fetchComments() {
    const response = await axios.get<string>('/api');
    /** @pact-response-body */
    const data = response.data;
    // ...
}
```
**IMPORTANT** - JSDoc has to be applied to separate variable - **not** directly to axios response
```ts
    /** @pact-response-body */ - WRONG!;
    const response = await axios.get<string>('/api');
    /** @pact-response-body */ - CORRECT;
    const data = response.data;
```
````

We should give firstly some valid example of this annotation, and then give some additional information

File: README.md [hunk 2]
```diff
@@ -1,75 +1,163 @@
...
+| `provider`         | Provider's name                                                                                                                                                                         |
+| `files`            | Array of glob patterns where API functions are defined                                                                                                                                  |
+| `requestHeaders`   | Request headers shared across all requests                                                                                                                                              |
+| `responseHeaders`  | Response headers shared across all responses                                                                                                                                            |
+| `queryArrayFormat` | Sets separator for array in query - possible options are: `indices`, `brackets`, `comma` and `repeat` [source](https://github.com/ljharb/qs#stringifying). Default value is `brackets`. |
```
Comment (Fiszcz) [inline README.md:110]: Could we take "source" in parentheses? (source).

**Review by Fiszcz — APPROVED** [review]

**Comment by Fiszcz** [comment]
Zajebista robota!

File: README.md [hunk 3]
```diff
@@ -1,75 +1,163 @@
...
+If you're limited to historical versions of TypeScript, you should install a corresponding version of pact-gen-ts. Below table presents what TS versions pact-gen-ts will work with:
+
+| pact-gen-ts | TypeScript |
+| ----------- | ---------- |
+| 0.8         | 4.1 - 4.2  |
+| 0.9 - x     | \>= 4.5    |
+
+## Configuration
+
+Pact-gen-ts uses configuration stored in `pacts.config.js` file in project's root directory:
```
[the remark this answers is not in the document]
Comment (pawfa) [reply README.md:70]: Regarding those genitives - before adding I've checked them in google :) - I think it is correct but will verify it with native speaker.

File: README.md [hunk 4]
```diff
@@ -1,75 +1,163 @@
...
 
-## [x] Add metadata and pact specification into pacts
+| Option      | Description                                                                            |
+| ----------- | -------------------------------------------------------------------------------------- |
+| `consumer`  | Consumer's name                                                                        |
+| `buildDir`  | Directory where generated pacts will be placed                                         |
+| `verbose`   | If set to `true` additional information during pacts generating process will be logged |
+| `providers` | Array of options for each provider                                                     |
 
-Pact Specification in version 2.0.0
```
Comment (lukaszfiszer) [inline README.md:49]: Quick idea - document `providers` in the same table as other options using dot-notation for increased readability.


| Option      |  Required  |   Default   | Description                                                                            |
|-------------|:----------:|:-----------:|----------------------------------------------------------------------------------------|
| `consumer`  |    Yes     |      -      | Consumer's name                                                                        |
| `providers` |    Yes     |      -      | Array of options for each provider                                                     |
| `providers[].provider` |    Yes     |      -      |  Provider's name                                                     |
| `providers[].files` |    Yes     |      -      |    Array of glob patterns where API functions are defined                                                   |
| `providers[]...` |    Yes     |      -      |                                                       |
| `buildDir`  |     No     |  `./pacts`  | Directory where generated pacts will be placed                                         |
| `verbose`   |     No     |   `false`   | If set to `true` additional information during pacts generating process will be logged

**Review by lukaszfiszer — APPROVED** [review]
Awesome job!

File: README.md [hunk 5]
```diff
@@ -1,75 +1,163 @@
...
         },
     ],
 };
 ```
 
-### [x] Specify multiple providers
+### Options
 
-## [x] Add metadata and pact specification into pacts
+| Option      | Description                                                                            |
```
[the remark this answers is not in the document]
Comment (Fiszcz) [reply README.md:95]: Yeah, this looks fine

File: README.md [hunk 6]
```diff
@@ -81,225 +169,72 @@ function fetchComments() {
...
  */
 function fetchComments() {
     // ...
 }
 ```
 
-## [x] Specify response status of interaction
+`@pact-path` - sets path
 
 ```ts
```
[the remark this answers is not in the document]
Comment (pawfa) [reply README.md:180]: I removed any descriptions regarding required/optional annotations. They are necessary for pact verification, not for pact generation.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pawfa) [reply README.md:70]: Asked for review native English speaker - luckily no problems found.

**Pull request merged**